libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -417,6 +417,46 @@ def ListDiskByLabels(self,
     return self._ListByLabel(
         labels_filter, disk_service_object, filter_union)
 
+  def ListReservedExternalIps(self, zone: str) -> List[str]:
+    """Lists all static external IP addresses that are available to a zone.
+
+    The method first converts the zone to a region,
+    and then queries the GCE addresses resource.
+
+    Args:
+      zone (str): The zone in which the returned IPs would be available.
+
+    Returns:
+      List[str]: The list of available IPs in the specified zone.
+
+    Raises:
+      ValueError: If the zone is malformed.
+      errors.ResourceNotFoundError: If the request did not succeed.
+    """
+    # Convert zone to region
+    zone_parts = zone.split('-')
+    if len(zone_parts) != 3:
+      raise ValueError('Invalid zone: {0:s}.'.format(zone))
+    region = '-'.join(zone_parts[:-1])
+    # Request list of addresses
+    addresses_client = self.GceApi().addresses()
+    request = addresses_client.list(project=self.project_id, region=region)
+    try:
+      response = request.execute()
+    except HttpError as exception:
+      message = 'Unable to list external IPs for {0:s}: {1:s}'.format(
+        self.project_id,
+        exception.error_details)
+      raise errors.ResourceNotFoundError(message, __name__)
```

Comment:
better use ExecuteRequest that will handle errors and paging than adding try/except block for everu call, also paging isn't handled that way. Keep in mind that you will get a list of responses that you need to loop on https://github.com/google/cloud-forensics-utils/blob/a849b99e780726969fd3fecf8ccab1d4ddcc5ebd/libcloudforensics/providers/gcp/internal/common.py#L265

Reply:
It seems that ExecuteRequest only handles errors with authentication; if there is any HttpError it will not be handled. It will instead be passed up to the caller so the try/catch block might still be necessary here.

Reply:
We might want to catch HttpError directly in `ExecuteRequest` in a separate block

Reply:
Okay, so I think this calls for a couple enhancements to `ExecuteRequest`:

1. We catch the `HttpError` directly in the call, which would require an additional argument as to which exception to raise
2. We also need a way to do requests that only require a single response, and logs warnings in the case that there are more than one response

My suggestion would be to merge this PR, and then consider this refactoring a separate issue.

Reply:
1) Actually I _think_ we might have left out `HttpError` from `ExecuteRequest` voluntarily back then: catching the `HttpError` directly from the caller allows us to raise the proper error subsequently (e.g. here `ResourceNotFoundError`). So, scratch that; you can leave it as is, sorry for my earlier comment.

2) Imo it's okay to keep returning a list of responses, and handle the logic in the caller. `ExecuteRequest` is just a wrapper around executing the API call, and I don't think it should handle anything more than that (i.e. any logic on the responses themselves, including how many responses were returned or how they are formatted).